### PR TITLE
Feature/foodtrucks

### PR DESCRIPTION
**Pull request by thiteixeira**
Basic functionality of the food truck intent is working, i.e., it tells the user how many food trucks there are in the vicinity.

This PR closes issue #52 

PS: I added a module on the requirements file to convert Point X_Y coordinates to Lat/Long

### CODE REVIEW

**Comment by curranjm** [comment]
@thiteixeira, it looks like you have a lot of code in the diff that's unrelated to the food trucks intent. Trying pulling master into your branch and we'll be able to see the code specific to this intent.

**Comment by thiteixeira** [comment]
Hey @curranjm 

Most of the files are necessary:
`mycity/mycity/deploy_tools/requirements.txt` uses a library for coordinates conversion
`mycity/mycity/intents/food_truck_intent.py` the intent itself
`mycity/mycity/logger.py` is the consolidated logger that hasn't been merged yet. It has no effect on other files though
`mycity/mycity/mycity_controller.py` includes the foodtruck intent
`mycity/mycity/utilities/gis_utils.py` adds a couple of methods to translate address to lat/lon and vice versa
`mycity/platforms/amazon/models/en_US.json` adds the utterances for this intent (though I haven't changed this file)

Is it possible to merge just these files?

**Comment by curranjm** [comment]
My mistake, I see. I'll look through this before Tuesday.

File: mycity/platforms/amazon/models/en_US.json [hunk 1]
```diff
@@ -1133,6 +1133,15 @@
...
                 },
+                {
+                    "name": "FoodTruckIntent",
+                    "slots": [],
+                    "samples": [
+                        "food trucks",
+                        "foodtrucks",
+                        "Nearby foodtrucks"
+                    ]
+                },
```
Comment (jmartini) [inline mycity/platforms/amazon/models/en_US.json]: This causes errors when uploading to the Alexa developer console. I think this is in the wrong section and should  be in the `intents` array above.

**Comment by jmartini** [comment]
This currently only returns the number of trucks in the area. I think it would be better to also list those trucks either right away or if the user asks for more info.

**Comment by thiteixeira** [comment]
Hi @jmartini Hi @curranjm 
I will work on the comments sometime before next Tuesday.
We can return the closest three food trucks and tell the user that there are other N food trucks around (with an option to read the next 3 or 5).
Let me know if you have any other comments

File: mycity/mycity/intents/food_truck_intent.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                "I can't seem to find {}. Try another address" \
+                    .format(address_string)
+            mycity_response.reprompt_text = None
+            mycity_response.session_attributes = mycity_request.session_attributes
+            mycity_response.card_title = "Food Trucks"
+            mycity_request = clear_address_from_mycity_object(mycity_request)
+            mycity_response = clear_address_from_mycity_object(mycity_response)
+            return mycity_response
+
+        except BadAPIResponse:
```
[the remark this answers is not in the document]
Comment (thiteixeira) [reply mycity/mycity/intents/food_truck_intent.py]: should be fixed now

File: mycity/mycity/intents/food_truck_intent.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        except InvalidAddressError:
+            address_string = address
+
+            mycity_response.output_speech = \
+                "I can't seem to find {}. Try another address" \
+                    .format(address_string)
+            mycity_response.reprompt_text = None
+            mycity_response.session_attributes = mycity_request.session_attributes
+            mycity_response.card_title = "Food Trucks"
+            mycity_request = clear_address_from_mycity_object(mycity_request)
```
[the remark this answers is not in the document]
Comment (thiteixeira) [reply mycity/mycity/intents/food_truck_intent.py]: should be fixed now

File: mycity/mycity/intents/food_truck_intent.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
+"""
```
[the remark this answers is not in the document]
Comment (thiteixeira) [reply mycity/mycity/intents/food_truck_intent.py]: I pushed a new `food_truck_intent` that keeps the line length at maximum 80

**Comment by thiteixeira** [comment]
Food truck intent is working and it was tested against [City of Boston Food Truck Schedule](https://www.boston.gov/departments/small-business-development/city-boston-food-trucks-schedule).

This commit is still missing the dialog, so in the case that there are more than three food trucks, the user will be able to request for more entries (4th food truck on the list, 5th food truck, and so on).

**Comment by jmartini** [comment]
@thiteixeira I tried running this and ran into some errors in the Alexa test console. Are you able to run this? Is there anything you forgot to commit or push up?

**Comment by thiteixeira** [comment]
Yes, it works well on my machine. There's nothing else to commit on my end.
What errors are you getting?

**Comment by thiteixeira** [comment]
Moving discussion to PR #190

**Pull request closed without merging**